pkg/plugins/guardian/guardian.go

```diff
@@ -0,0 +1,85 @@
+package guardian
```

Comment:
Note that this guardian will be removed or largely simplefied once we introduce `plugins:read` and `plugins.settings:read` as knowing if a user can view a plugin and its settings will be a direct check and not derived from other permissions.